src/main/Dashboard.java

```diff
@@ -1,10 +1,131 @@
+            System.out.println("[4] DELETE USER");
+
+            try {
+                System.out.println("Choose a menu: ");
+
+                while (!scanner.hasNextInt()) {
+                    scanner.next();
+                }
+
+                menu = scanner.nextInt();
+                scanner.nextLine();
+
+                switch (menu) {
+                case 0:
+                    // Quit
+                    System.out.println("GoodBye...");
+                    System.exit(0);
+                    break;
+
+                case 1:
+                    // Register User
+                    registerUser();
+                    break;
+
+                case 2:
+                    // List out all Users
+                    listUsers();
+                    break;
+
+                case 3:
+                    // Modify a User
+                    System.out.println("Please enter the current email of the user you wish to modify:");
+                    email = scanner.next().toString();
+                    if (findUser(email)) {
+                        deleteUser(email);
+                        scanner.nextLine();
+                        registerUser();
+                        System.out.println("User details modified:");
+
+                    } else {
+                        System.out.println("This user does not exist \n \n");
+                    }
+
+                    break;
+
+                case 4:
+                    // Delete a User
+                    System.out.println("Please enter the current email of the user you wish to delete:");
+                    email = scanner.next().toString();
+                    deleteUser(email);
+                    break;
+
+                default:
+                    System.out.println("You have entered an valid menu :( \nPlease try again");
+                    break;
+                }
+            } catch (InputMismatchException e) {
+                System.out.println("Incorrect menu entered :( \nPlease try again");
+            }
+
+        } while (true);
+
+    }
+
+    public static void registerUser() {
+        User user = new User();
+        System.out.println("Please enter name:");
+        user.setName(scanner.nextLine());
+        try {
+            System.out.println("Please enter age:");
+            user.setAge(scanner.nextInt());
+        } catch (InputMismatchException e) {
+            System.out.println("Incorrect age entered! \n");
+        }
+        System.out.println("Please enter email:");
+        user.setEmail(scanner.next());
+        insertUser(user);
+    }
+
+    public static void insertUser(User user) {
+        hashUsers.add(user);
+    }
+
+    public static void listUsers() {
+        for (User element : hashUsers) {
+            System.out.println(element + "\n");
+        }
+    }
+
+    public static boolean findUser(String email) {
+        boolean exists = false;
+        for (User element : hashUsers) {
```

Comment:
This method does what it's supposed to do, but in these situations (when you are searching for an object inside an array or collection) it's better to shortcut the `for` loop when you found the desired element.

Consider the following:
`hashUsers` contains 5.000.000 `User` objects. The second one that you check fits your email criteria, so you essentially found what you were looking for. But your method does not stop there and return the found object; instead it moves on and iterates over all the 5.000.000 elements. Could be an impact on performance.

This is actually one of the situations where the "one return statement principle" is not the right approach.